modules/extension/wmts/src/main/java/org/geotools/map/WMTSMapLayer.java

```diff
@@ -0,0 +1,142 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2017, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.map;
+
+import java.util.logging.Logger;
+
+import org.geotools.data.ows.Layer;
+import org.geotools.data.wms.WebMapServer;
+import org.geotools.data.wmts.WebMapTileServer;
+import org.geotools.data.wmts.request.GetTileRequest;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.referencing.CRS;
+import org.geotools.styling.FeatureTypeStyle;
+import org.geotools.styling.RasterSymbolizer;
+import org.geotools.styling.Rule;
+import org.geotools.styling.Style;
+import org.geotools.styling.StyleFactory;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+/**
+ * Wraps a WMTS layer into a {@link MapLayer} for interactive rendering usage
+ * 
+ * TODO: expose a GetFeatureInfo that returns a feature collection
+ * TODO: expose the list of named styles and allow choosing which style to use
+ *
+ * @author Ian Turton
+ * @author Emanuele Tajariol (etj at geo-solutions dot it)
+ */
+public class WMTSMapLayer extends GridReaderLayer {
+
+    static public final Logger LOGGER = org.geotools.util.logging.Logging
+            .getLogger("org.geotools.map");
+    /**
+     * The default raster style
+     */
+    final static Style STYLE;
+    static {
+        StyleFactory factory = CommonFactoryFinder.getStyleFactory(null);
+        RasterSymbolizer symbolizer = factory.createRasterSymbolizer();
+
+        Rule rule = factory.createRule();
+        rule.symbolizers().add(symbolizer);
+
+        final FeatureTypeStyle type = factory.createFeatureTypeStyle();
+        type.rules().add(rule);
+
+        STYLE = factory.createStyle();
+        STYLE.featureTypeStyles().add(type);
+    }
+
+    private String rawTime;
+
+    /**
+     * Builds a new WMS layer
+     *
+     * @param wmts
+     * @param layer
+     */
+    public WMTSMapLayer(WebMapTileServer wmts, Layer layer) {
+        super( new WMTSCoverageReader(wmts, layer), STYLE );
```

Comment:
Danger danger, Style objects are mutable, if someone gets the style and changes it, it will affect the static variable and all subsequent requests. Better create the style every time.